src/Facade/Services/StoresPalletFacadeService.cs

```diff
@@ -118,7 +118,9 @@ protected override async Task UpdateFromResourceAsync(
 
         protected override Expression<Func<StoresPallet, bool>> SearchExpression(string searchTerm)
         {
-            throw new NotImplementedException();
+            bool isInt = int.TryParse(searchTerm, out int palletNumber);
+
+            return x => (isInt && x.PalletNumber == palletNumber) || x.Description.ToUpper().Contains(searchTerm.Trim().ToUpper());
```

Comment:
I'm not sure how much sense it makes to search pallet numbers like this. Its just a number - you either know it or you dont.